conda/cli/common.py

```diff
@@ -376,7 +376,7 @@ def validate_environment_files_consistency(files: list[str]) -> None:
 
     # Get types for all files using the plugin manager
     file_types = {
-        file: context.plugin_manager.get_environment_specifier_name(file)
+        file: context.plugin_manager.get_environment_specifier(file).name
```

Comment:
https://github.com/conda/conda/pull/14725/ also introduced some changes to the plugin manager that are at odds with the changes made to the plugin manager in this PR. After rebasing, I've straighened this out to be consistent with what we chatted about here https://github.com/conda/conda/pull/14877/files#r2109000518. This change is a side effect of this.